packages/salesforcedx-vscode-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -53,13 +100,152 @@ export abstract class BaseNode extends TreeItem {
+    // Get the executableEntityName from the sourceInput, if this is a trigger
+    // then ensure that the SFDC_TRIGGER prefix is prepended to the trigger name
+    if (sourceNameSplit[1].toLocaleLowerCase() === 'trigger') {
+      this.executableEntityName = SFDC_TRIGGER + sourceNameSplit[0];
+      this.isTrigger = true;
+    } else {
+      this.executableEntityName = sourceNameSplit[0];
+      this.isTrigger = false;
+    }
+    this.isDumpingHeap = true;
+
+    // Create the items that the user is going to be able to control (Type, Script, Iteration)
+    const cpASTNode = new CheckpointInfoActionScriptTypeNode(
+      ActionScriptType.None
+    );
+    this.children.push(cpASTNode);
+    const cpScriptNode = new CheckpointInfoActionScriptNode('');
+    this.children.push(cpScriptNode);
+    const cpIterationNode = new CheckpointInfoIterationNode(1);
+    this.children.push(cpIterationNode);
+  }
+
+  public createJsonStringForApexExecutionOverlayAction(): string {
+    const jsonNameValueSeperator = ', ';
+    // Start the JSON name/value pairs with the executable entity name
+    let returnString = CheckpointUtil.formatNameValuePairForJSon(
+      APEX_EXECUTION_OVERLAY_ACTION_EXECUTABLE_ENTITY_NAME,
+      this.executableEntityName
+    );
+    returnString += jsonNameValueSeperator;
+    // Add all of the children
+    for (const cpInfoNode of this.children) {
+      returnString += cpInfoNode.createJsonStringForApexExecutionOverlayAction();
+      returnString += jsonNameValueSeperator;
+    }
+    // Add isDumpingHeap
+    returnString += CheckpointUtil.formatNameValuePairForJSon(
+      APEX_EXECUTION_OVERLAY_ACTION_IS_DUMPING_HEAP,
+      this.isDumpingHeap
+    );
+    returnString += jsonNameValueSeperator;
+    // Add line
+    returnString += CheckpointUtil.formatNameValuePairForJSon(
+      APEX_EXECUTION_OVERLAY_ACTION_LINE,
+      this.line
+    );
+    return returnString;
+  }
 
   public getChildren(): CheckpointInfoNode[] {
     return this.children;
   }
 }
 
-export abstract class CheckpointInfoNode extends BaseNode {
+export class CheckpointInfoNode extends BaseNode {
+  public createJsonStringForApexExecutionOverlayAction(): string {
+    return '';
+  }
+  public getChildren(): BaseNode[] {
+    return [];
+  }
+}
+
+// Specialty node defintions. These are for nodes that the user is capable of
+// changing the values for and are of different value types:
+// CheckpointInfoActionScriptNode: holds the action script (string)
+// CheckpointInfoActionScriptTypeNode: holds the type of action script (ActionScriptType enum)
+// CheckpointInfoIterationsNode: holds the number of iterations (integer)
+// The reason for creating these would be to add specific validations when
+// the fields are made editable.
+export class CheckpointInfoActionScriptNode extends CheckpointInfoNode {
+  public actionScript: string;
+  constructor(actionScriptInput: string) {
+    super('Script: ' + actionScriptInput);
```

Comment:
Can these labels be externalized?

Reply:
These shouldn't be needed outside of checkpointService.ts, I can make them constants in there, which will at least make them defined in one place and easier to find.